core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,116 @@
+        try {
+            pwFile = CharStreams.writingStringToFilePrintWriter(TODO_LIST_PW_FILE , TODO.toString());
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingTextData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingStringToFileFileWriter(TODO_LIST_FW_FILE, TODO.toString());
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+        
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFilePrintWriter(TODO_LIST_PW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingFormattedData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingNonStringToFileFileWriter(TODO_LIST_FW_FILE, TODO_1);
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_1.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenPrintWriterAppendsToFile() throws IOException {
+        try {
+            CharStreams.appendingToFilePrintWriter(TODO_LIST_PW_FILE, TODO_2);
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenWritingToExixtingFile_thenFileWriterAppendsToFile() throws IOException {
+        try {
+            CharStreams.appendingToFileFileWriter(TODO_LIST_FW_FILE, TODO_2);
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_2.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenCheckingError_thenPrintWriterReturnsFalse() throws IOException {
+        try {
+            boolean result = CharStreams.printWriterErrorFlag(TODO_LIST_PW_FILE, TODO_2);
+            assertFalse(result);
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenHandlingError_thenFileWriterHandlesError() {
+        try (FileWriter fileWriter = new FileWriter(TODO_LIST_FW_FILE, true)) {
+        fileWriter.write(TODO_2.toString());
+        } catch (IOException e) {
+            fail("error not expected", e);
+        }
+        assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO_2.toString());

```

Comment:
The assertion was made on the wrong file and that is why the test is failing

```suggestion
        assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO_2.toString());

```